core/app/models/spree/product.rb

```diff
@@ -161,6 +161,11 @@ def discontinued?
       !!discontinue_on && discontinue_on <= Time.current
     end
 
+    # determine if any variant (including master) can be supplied
+    def can_supply?
+      variants_including_master.any?(&:can_supply?)
```

Comment:
@priyank-gupta Ah, then it's ok. I was testing it on spree sandbox application and there master variant was still 10.